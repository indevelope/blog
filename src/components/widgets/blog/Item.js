import React, { PropTypes } from 'react';

import { Image, Item } from 'semantic-ui-react';

import Meta from './elements/Meta';

const BlogItem = ({ item }) => (
  <Item>
    <Image {...item.image} />
    <Item.Content>
      <Item.Header>{item.title}</Item.Header>
      <Meta {...item.meta} />
    </Item.Content>
  </Item>
);

BlogItem.propTypes = {
  item: PropTypes.object
};

export default BlogItem;
