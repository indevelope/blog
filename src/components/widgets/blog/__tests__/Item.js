import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import Link from 'components/elements/Link';

import Item from '../Item';

describe('Item', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Item />, div);
  });

  describe('render', () => {
    it('should render the title', () => {
      const itemProps = { title: 'Hello, World!', id: 5 };

      const item = shallow(<Item item={itemProps} />);
      const header = <Link to="/posts/5">Hello, World!</Link>;

      expect(item.contains(header)).toEqual(true);
    });

    it('should render usual item', () => {
      const itemProps = {
        title: 'Hello, World!',
        id: 5,
        image: {
          alt: 'Hello, World',
          src: '/images/World.jpg'
        }
      };

      const item = shallow(<Item item={itemProps} />);

      expect(item).toMatchSnapshot();
    })
  });
});
