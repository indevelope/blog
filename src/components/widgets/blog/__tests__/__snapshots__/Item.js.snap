// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item render should render usual item 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Item>
    <Image
        alt="Hello, World"
        as="img"
        src="/images/World.jpg"
        ui={true}
    />
    <ItemContent>
        <ItemHeader>
            <Link
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/posts/5"
            >
                Hello, World!
            </Link>
        </ItemHeader>
        <Meta />
    </ItemContent>
</Item>,
  "nodes": Array [
    <Item>
      <Image
            alt="Hello, World"
            as="img"
            src="/images/World.jpg"
            ui={true}
      />
      <ItemContent>
            <ItemHeader>
                  <Link
                        onlyActiveOnIndex={false}
                        style={Object {}}
                        to="/posts/5"
                  >
                        Hello, World!
                  </Link>
            </ItemHeader>
            <Meta />
      </ItemContent>
</Item>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <BlogItem
        item={
                Object {
                        "id": 5,
                        "image": Object {
                          "alt": "Hello, World",
                          "src": "/images/World.jpg",
                        },
                        "title": "Hello, World!",
                      }
        }
/>,
      "_debugID": 35,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "item": Object {
            "id": 5,
            "image": Object {
              "alt": "Hello, World",
              "src": "/images/World.jpg",
            },
            "title": "Hello, World!",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Item>
          <Image
                    alt="Hello, World"
                    as="img"
                    src="/images/World.jpg"
                    ui={true}
          />
          <ItemContent>
                    <ItemHeader>
                              <Link
                                        onlyActiveOnIndex={false}
                                        style={Object {}}
                                        to="/posts/5"
                              >
                                        Hello, World!
                              </Link>
                    </ItemHeader>
                    <Meta />
          </ItemContent>
</Item>,
        "_debugID": 36,
        "_renderedOutput": <Item>
          <Image
                    alt="Hello, World"
                    as="img"
                    src="/images/World.jpg"
                    ui={true}
          />
          <ItemContent>
                    <ItemHeader>
                              <Link
                                        onlyActiveOnIndex={false}
                                        style={Object {}}
                                        to="/posts/5"
                              >
                                        Hello, World!
                              </Link>
                    </ItemHeader>
                    <Meta />
          </ItemContent>
</Item>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <BlogItem
    item={
        Object {
            "id": 5,
            "image": Object {
              "alt": "Hello, World",
              "src": "/images/World.jpg",
            },
            "title": "Hello, World!",
          }
    }
/>,
}
`;
